rust: Ensure cargo executable path is fetched for custom targets

Yashinde145 opened the pull request:
Fix the cargo binary path error and ensure that it is fetched during rustc bootstrap on custom targets.

```
=========================================
 ERROR: test_cargoflags (bootstrap_test.BuildBootstrap)
----------------------------------------------------------------------
  Traceback (most recent call last):
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 157, in test_cargoflags
     args, _ = self.build_args(env={"CARGOFLAGS": "--timings"})
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 154, in build_args
     return build.build_bootstrap_cmd(env), env
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap.py", line 960, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at /home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo
```

Edit(albertlarsan68): Add code fences

Yashinde145 committed:
rust: Ensure cargo executable path is fetched for custom targets

Fix the cargo binary path error and ensure that it is fetched
during rustc bootstrap on custom targets.

edit: remove "cargo" from os.path.join

======================================================================
 ERROR: test_cargoflags (bootstrap_test.BuildBootstrap)
----------------------------------------------------------------------
  Traceback (most recent call last):
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 157, in test_cargoflags
     args, _ = self.build_args(env={"CARGOFLAGS": "--timings"})
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 154, in build_args
     return build.build_bootstrap_cmd(env), env
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap.py", line 960, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`

Signed-off-by: Yash Shinde <uayashu145@gmail.com>

Yashinde145 committed:
rust: Ensure cargo executable path is fetched for custom targets

Fix the cargo binary path error and ensure that it is fetched
during rustc bootstrap on custom targets.

======================================================================
 ERROR: test_cargoflags (bootstrap_test.BuildBootstrap)
----------------------------------------------------------------------
  Traceback (most recent call last):
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 157, in test_cargoflags
     args, _ = self.build_args(env={"CARGOFLAGS": "--timings"})
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 154, in build_args
     return build.build_bootstrap_cmd(env), env
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap.py", line 960, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`

Signed-off-by: Yash Shinde <uayashu145@gmail.com>

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7581707268/job/20649838345) [(plain)](https://github.com/rust-lang/rust/commit/fea0f15b1b31eba0e266ec25b7470ded1e17a84e/checks/20649838345/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=Yashinde145
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_7294416d-84de-4e4b-af6e-d1c28fb9147a
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=964820413aa987de134062330e57b2b5e861e20b
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_7294416d-84de-4e4b-af6e-d1c28fb9147a
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_7294416d-84de-4e4b-af6e-d1c28fb9147a
GITHUB_TRIGGERING_ACTOR=Yashinde145
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/120125/merge
GITHUB_WORKFLOW_SHA=964820413aa987de134062330e57b2b5e861e20b
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
Built container sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Looks like docker image is the same as before, not uploading
https://ci-caches.rust-lang.org/docker/7ebc15c01a233894034d277c8cce4e949f4e7791f66b4727c8fb6e058a0b8171d6152e1441d677cef0653843ceeee469c097b8699b2bb74249e674f6aa1a8813
sha256:9c3c93a371e5aed5c18185b24f130d95d5140dbd72a9b325e7b6b49e521a4faa
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Fri Jan 19 09:12:19 UTC 2024
  network time: Fri, 19 Jan 2024 09:12:19 GMT
  network time: Fri, 19 Jan 2024 09:12:19 GMT
##[endgroup]
sccache: Starting the server...
Traceback (most recent call last):
  File "/checkout/src/bootstrap/configure.py", line 12, in <module>
    import bootstrap # noqa: E402
  File "/checkout/src/bootstrap/bootstrap.py", line 963
  File "/checkout/src/bootstrap/bootstrap.py", line 963
    cargo_bin_path = os.path.join(self.bin_root(), "bin", "cargo")
TabError: inconsistent use of tabs and spaces in indentation
  network time: Fri, 19 Jan 2024 09:12:19 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

Yashinde145 committed:
Ensure cargo executable path is fetched for custom targets
Fix the cargo binary path error and ensure that it is fetched
during rustc bootstrap on custom targets.

======================================================================
 ERROR: test_cargoflags (bootstrap_test.BuildBootstrap)
----------------------------------------------------------------------
  Traceback (most recent call last):
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 157, in test_cargoflags
     args, _ = self.build_args(env={"CARGOFLAGS": "--timings"})
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap_test.py", line 154, in build_args
     return build.build_bootstrap_cmd(env), env
  File "/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/src/bootstrap/bootstrap.py", line 960, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/home/build-st/tmp/work/cortexa57-poky-linux/rust/1.74.1/rustc-1.74.1-src/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`

Signed-off-by: Yash Shinde <uayashu145@gmail.com>

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7581842002/job/20650220797) [(plain)](https://github.com/rust-lang/rust/commit/cd325561011c54e72df2dffe14c20f4ebc24eeea/checks/20650220797/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
GITHUB_ACTION=__run_7
GITHUB_ACTIONS=true
GITHUB_ACTION_REF=
GITHUB_ACTION_REPOSITORY=
GITHUB_ACTOR=Yashinde145
GITHUB_API_URL=https://api.github.com
GITHUB_BASE_REF=master
GITHUB_ENV=/home/runner/work/_temp/_runner_file_commands/set_env_b5f1edf1-c1f9-453d-957a-83b3e37fa620
GITHUB_EVENT_NAME=pull_request
---
GITHUB_SERVER_URL=https://github.com
GITHUB_SHA=cea92d75418097495c6f3bc60baa421830fab235
GITHUB_STATE=/home/runner/work/_temp/_runner_file_commands/save_state_b5f1edf1-c1f9-453d-957a-83b3e37fa620
GITHUB_STEP_SUMMARY=/home/runner/work/_temp/_runner_file_commands/step_summary_b5f1edf1-c1f9-453d-957a-83b3e37fa620
GITHUB_TRIGGERING_ACTOR=Yashinde145
GITHUB_WORKFLOW_REF=rust-lang/rust/.github/workflows/ci.yml@refs/pull/120125/merge
GITHUB_WORKFLOW_SHA=cea92d75418097495c6f3bc60baa421830fab235
GITHUB_WORKSPACE=/home/runner/work/rust/rust
GOROOT_1_19_X64=/opt/hostedtoolcache/go/1.19.13/x64
---
##[group]Building bootstrap
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1107, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 968, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 2/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1140, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1107, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 968, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 3/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in <module>
##[group]Building bootstrap
##[group]Building bootstrap
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1140, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1107, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 968, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 4/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1140, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1107, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 968, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 5/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1140, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1107, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 968, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
The command has failed after 5 attempts.
  local time: Fri Jan 19 09:24:30 UTC 2024
  network time: Fri, 19 Jan 2024 09:24:30 GMT
##[error]Process completed with exit code 1.
```

</details>

Yashinde145 committed:
Merge branch 'rust-lang:master' into cargo_path

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7695040364/job/20967090301) [(plain)](https://github.com/rust-lang/rust/commit/dd4a3ec1be5a2c1d84f38fa5c9a285d6e3993515/checks/20967090301/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#12 writing image sha256:8ab754459a4c7f9f42eb506c6950fdca3beb8ade26e3ad633af45ef00f60c091 done
#12 naming to docker.io/library/rust-ci done
#12 DONE 10.2s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Mon Jan 29 11:06:19 UTC 2024
  network time: Mon, 29 Jan 2024 11:06:19 GMT
  network time: Mon, 29 Jan 2024 11:06:19 GMT
##[endgroup]
sccache: Starting the server...
##[group]Configure the build
configure: processing command line
configure: 
configure: build.configure-args := ['--build=x86_64-unknown-linux-gnu', '--llvm-root=/usr/lib/llvm-16', '--enable-llvm-link-shared', '--set', 'rust.thin-lto-import-instr-limit=10', '--set', 'change-id=99999999', '--enable-verbose-configure', '--enable-sccache', '--disable-manage-submodules', '--enable-locked-deps', '--enable-cargo-native-static', '--set', 'rust.codegen-units-std=1', '--set', 'dist.compression-profile=balanced', '--dist-compression-formats=xz', '--set', 'build.optimized-compiler-builtins', '--disable-dist-src', '--release-channel=nightly', '--enable-debug-assertions', '--enable-overflow-checks', '--enable-llvm-assertions', '--set', 'rust.verify-llvm-ir', '--set', 'rust.codegen-backends=llvm,cranelift,gcc', '--set', 'llvm.static-libstdcpp', '--enable-new-symbol-mangling']
configure: target.x86_64-unknown-linux-gnu.llvm-config := /usr/lib/llvm-16/bin/llvm-config
configure: llvm.link-shared     := True
configure: rust.thin-lto-import-instr-limit := 10
configure: change-id            := 99999999
---
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 2/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
    main()
    main()
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 3/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
##[group]Building bootstrap
##[group]Building bootstrap
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 4/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
Command failed. Attempt 5/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
    main()
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
##[group]Building bootstrap
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
  File "/checkout/src/bootstrap/bootstrap.py", line 983, in build_bootstrap_cmd
    raise Exception("no cargo executable found at `{}`".format(
Exception: no cargo executable found at `/checkout/obj/build/x86_64-unknown-linux-gnu/stage0/bin/cargo`
The command has failed after 5 attempts.
  local time: Mon Jan 29 11:06:38 UTC 2024
  network time: Mon, 29 Jan 2024 11:06:38 GMT
##[error]Process completed with exit code 1.
```

</details>

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-16`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7695402142/job/20968186707) [(plain)](https://github.com/rust-lang/rust/commit/0019bfc65e3b0b7ce1af673235b18fc19fba942f/checks/20968186707/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
#12 writing image sha256:39cc9b02509f1879dbe37551670dedf0c0891345f9c80ee7ec21525ba513b390 done
#12 naming to docker.io/library/rust-ci done
#12 DONE 10.0s
##[endgroup]
Setting extra environment values for docker:  --env ENABLE_GCC_CODEGEN=1 --env GCC_EXEC_PREFIX=/usr/lib/gcc/
[CI_JOB_NAME=x86_64-gnu-llvm-16]
##[group]Clock drift check
  local time: Mon Jan 29 11:36:11 UTC 2024
  network time: Mon, 29 Jan 2024 11:36:11 GMT
  network time: Mon, 29 Jan 2024 11:36:11 GMT
##[endgroup]
sccache: Starting the server...
Traceback (most recent call last):
  File "/checkout/src/bootstrap/configure.py", line 12, in <module>
    import bootstrap # noqa: E402
  File "/checkout/src/bootstrap/bootstrap.py", line 977
  File "/checkout/src/bootstrap/bootstrap.py", line 977
    cargo_bin_path = os.path.join(self.bin_root(), "bin") + \
TabError: inconsistent use of tabs and spaces in indentation
  network time: Mon, 29 Jan 2024 11:36:11 GMT
##[error]Process completed with exit code 1.
Post job cleanup.
```

</details>

Yashinde145 committed:
rust: Ensure cargo executable path is fetched for custom targets

Fix indentation

Signed-off-by: Yash Shinde <uayashu145@gmail.com>

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/7695664320/job/20969041393) [(plain)](https://github.com/rust-lang/rust/commit/ea6ba58830c1222965f7a6fda5a21cf5882115d8/checks/20969041393/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
ENV PATH="$NODE_FOLDER:${PATH}"

COPY host-x86_64/x86_64-gnu-tools/browser-ui-test.version /tmp/

# For now, we need to use `--unsafe-perm=true` to go around an issue when npm tries
# to create a new folder. For reference:
# https://github.com/puppeteer/puppeteer/issues/375
# We also specify the version in case we need to update it to go around cache limitations.
#
# The `browser-ui-test.version` file is also used by bootstrap to emit warnings in case
# the local version of the package is different than the one used by the CI.
---
python3 "$X_PY" test --stage 2 check-tools
python3 "$X_PY" test --stage 2 src/tools/clippy
python3 "$X_PY" test --stage 2 src/tools/rustfmt

# Testing Miri is a bit more complicated.
# We set the GC interval to the shortest possible value (0 would be off) to increase the chance
# that bugs which only surface when the GC runs at a specific time are more likely to cause CI to fail.
# This significantly increases the runtime of our test suite, or we'd do this in PR CI too.
if [[ -z "${PR_CI_JOB:-}" ]]; then
    MIRIFLAGS=-Zmiri-provenance-gc=1 python3 "$X_PY" test --stage 2 src/tools/miri
    python3 "$X_PY" test --stage 2 src/tools/miri
fi
# We natively run this script on x86_64-unknown-linux-gnu and x86_64-pc-windows-msvc.
# We natively run this script on x86_64-unknown-linux-gnu and x86_64-pc-windows-msvc.
# Also cover some other targets via cross-testing, in particular all tier 1 targets.
case $HOST_TARGET in
  x86_64-unknown-linux-gnu)
    # Only this branch runs in PR CI.
    # Fully test all main OSes, including a 32bit target.
    python3 "$X_PY" test --stage 2 src/tools/miri --target i686-pc-windows-msvc
    # Only run "pass" tests for the remaining targets, which is quite a bit faster.
    python3 "$X_PY" test --stage 2 src/tools/miri --target x86_64-pc-windows-gnu --test-args pass
    python3 "$X_PY" test --stage 2 src/tools/miri --target i686-unknown-linux-gnu --test-args pass
    python3 "$X_PY" test --stage 2 src/tools/miri --target i686-unknown-linux-gnu --test-args pass
    python3 "$X_PY" test --stage 2 src/tools/miri --target aarch64-unknown-linux-gnu --test-args pass
    python3 "$X_PY" test --stage 2 src/tools/miri --target s390x-unknown-linux-gnu --test-args pass
    ;;
  x86_64-pc-windows-msvc)
    # Strangely, Linux targets do not work here. cargo always says
    # "error: cannot produce cdylib for ... as the target ... does not support these crate types".
    # Only run "pass" tests, which is quite a bit faster.
    python3 "$X_PY" test --stage 2 src/tools/miri --target i686-pc-windows-gnu --test-args pass
    ;;
  *)
    echo "FATAL: unexpected host $HOST_TARGET"
---
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 2/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 3/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
    main()
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
##[group]Building bootstrap
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 4/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
##[group]Building bootstrap
##[group]Building bootstrap
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 5/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1170, in <module>
    main()
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1155, in main
##[group]Building bootstrap
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1122, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
The command has failed after 5 attempts.
  local time: Mon Jan 29 12:02:59 UTC 2024
  network time: Mon, 29 Jan 2024 12:02:59 GMT
##[error]Process completed with exit code 1.
```

</details>

Yashinde145 committed:
Merge branch 'rust-lang:master' into cargo_path

rust-log-analyzer commented:
The job **`x86_64-gnu-tools`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/8167846580/job/22328913427) [(plain)](https://github.com/rust-lang/rust/commit/cc3b873f4c309648c2f2ebac629720e3e5591923/checks/22328913427/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
  File "/checkout/src/bootstrap/bootstrap.py", line 1164, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1131, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 2/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1179, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1179, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1164, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1131, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 3/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1179, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1179, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1164, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1131, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 4/5:
Traceback (most recent call last):
##[group]Building bootstrap
  File "/checkout/src/bootstrap/bootstrap.py", line 1179, in <module>
  File "/checkout/src/bootstrap/bootstrap.py", line 1179, in <module>
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1164, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1131, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
Command failed. Attempt 5/5:
Traceback (most recent call last):
  File "/checkout/src/bootstrap/bootstrap.py", line 1179, in <module>
##[group]Building bootstrap
##[group]Building bootstrap
    main()
  File "/checkout/src/bootstrap/bootstrap.py", line 1164, in main
    bootstrap(args)
  File "/checkout/src/bootstrap/bootstrap.py", line 1131, in bootstrap
    build.build_bootstrap()
  File "/checkout/src/bootstrap/bootstrap.py", line 878, in build_bootstrap
    args = self.build_bootstrap_cmd(env)
  File "/checkout/src/bootstrap/bootstrap.py", line 980, in build_bootstrap_cmd
    cargo_bin_path = os.getenv("RUST_TARGET_PATH") + "rust-snapshot/bin/cargo"
TypeError: unsupported operand type(s) for +: 'NoneType' and 'str'
The command has failed after 5 attempts.
  local time: Wed Mar  6 06:25:41 UTC 2024
  network time: Wed, 06 Mar 2024 06:25:41 GMT
##[error]Process completed with exit code 1.
```

</details>
